Release_Process.md

```diff
@@ -42,7 +43,8 @@ These steps are common between minor and patch releases:
    - Target: the release branch that was just pushed
    - Title: Same as tag name
    - Description: The body of the changelog entry (Added, Changed etc)
-1. Publish the release on Github
+1. Publish the release on GitHub
+1. (Optional) Do a `make clean` in the repository root directory (`bigchaindb/`)
```

Comment:
@ttmc You may like a more compact approach via the `Makefile` as in https://github.com/bigchaindb/bigchaindb/pull/1165

We could then replace the three steps (`make clean, ..., twine upload`) by one: `make release`